src/lib/virtio/virtq_driver.lua

```diff
@@ -0,0 +1,247 @@
+-- Application to connect to a virtio-net driver implementation
+--
+-- Licensed under the Apache 2.0 license
+-- http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Copyright (c) 2015 Virtual Open Systems
+--
+
+module(..., package.seeall)
+
+local debug = _G.developer_debug
+
+local ffi    = require("ffi")
+local C      = ffi.C
+local memory = require('core.memory')
+local band   = require('bit').band
+
+local VirtioVirtq = {}
+VirtioVirtq.__index = VirtioVirtq
```

Comment:
It sets the base table itself as metatable for created instances. It is a common Lua idiom in many projects, though in Snabb I have seen the `return setmetatable(obj, { __index == BaseObject })` used more often (if not always).

It can be argued that setting the base object as metatable (`Base = {}; obj.__index = Base`) as done in the patch saves the creation of one table per instance created. In the particular case of Snabb applications, there will never be huge amounts of them being created, so the gain is marginal.